source/sigv4.c

```diff
@@ -351,45 +351,61 @@ static void setQueryParameterValue( size_t currentParameter,
                                     CanonicalContext_t * pCanonicalRequest );
 
 /**
- * @brief Update the HMAC using an input key.
+ * @brief Generates the key for the HMAC operation.
+ *
  * @note This function can be called multiple times before calling
- * #hmacData. Appending multiple substrings, then calling #hmacKey
- * on the appended string is also equivalent to calling #hmacKey on
+ * #hmacIntermediate. Appending multiple substrings, then calling #hmacAddKey
+ * on the appended string is also equivalent to calling #hmacAddKey on
  * each individual substring.
  * @note This function accepts a const char * so that string literals
  * can be passed in.
  *
  * @param[in] pHmacContext The context used for HMAC calculation.
  * @param[in] pKey The key used as input for HMAC calculation.
  * @param[in] keyLen The length of @p pKey.
+ * @param[in] isKeyPrefix Flag to indicate whether the passed key is
+ * prefix of a complete key for an HMAC operation. If this is a prefix,
+ * then it will be stored in cache for use with remaining part of the
+ * key that will be provided in a subsequent call to @ref hmacAddKey.
+ *
  * @return Zero on success, all other return values are failures.
  */
-static int32_t hmacKey( HmacContext_t * pHmacContext,
-                        const char * pKey,
-                        size_t keyLen );
+static int32_t hmacAddKey( HmacContext_t * pHmacContext,
+                           const char * pKey,
+                           size_t keyLen,
+                           bool isKeyPrefix );
 
 /**
- * @brief Update the HMAC using input data.
- * @note Must only be called after #hmacKey and leads to undefined
- * behavior otherwise. Likewise, one should not call #hmacKey after
- * calling #hmacData. One must call #hmacFinal first before calling
- * #hmacKey again.
+ * @brief Generates the intermediate hash output in the HMAC signing process.
+ * This represents the H( K' ^ i_pad || message ) part of the HMAC algorithm.
+ *
+ * @note This MUST be ONLY called after #hmacAddKey; otherwise results in
+ * undefined behavior. Likewise, one SHOULD NOT call #hmacAddKey after
+ * calling #hmacIntermediate. One must call #hmacFinal first before calling
+ * #hmacAddKey again.
  *
  * @param[in] pHmacContext The context used for HMAC calculation.
  * @param[in] pData The data used as input for HMAC calculation.
  * @param[in] dataLen The length of @p pData.
+ *
  * @return Zero on success, all other return values are failures.
  */
-static int32_t hmacData( HmacContext_t * pHmacContext,
-                         const char * pData,
-                         size_t dataLen );
+static int32_t hmacIntermediate( HmacContext_t * pHmacContext,
+                                 const char * pData,
+                                 size_t dataLen );
```

Comment:
The whole point of this naming scheme was to match the naming schemes of the hash interface.

Reply:
Also, the original terminology is more commonly used in cryptographic operations

Reply:
I felt that using terminology that follows the HMAC algorithmic operations is easier for a developer to make sense of the relationship between the `hmac*` internal functions. As these are internal functions, I prefer prioritize ease of readability

Reply:
It would be good to rename the hash interface then to match this new naming scheme

Reply:
It would be good to rename the hash interface then to match this new naming scheme